sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcDeserializer.scala

```diff
@@ -0,0 +1,304 @@
+        }
+
+      case BooleanType =>
+        new Converter {
+          override def set(value: Any): Unit =
+            updater.setBoolean(value.asInstanceOf[BooleanWritable].get)
+        }
+
+      case ByteType =>
+        new Converter {
+          override def set(value: Any): Unit = updater.setByte(value.asInstanceOf[ByteWritable].get)
+        }
+
+      case ShortType =>
+        new Converter {
+          override def set(value: Any): Unit =
+            updater.setShort(value.asInstanceOf[ShortWritable].get)
+        }
+
+      case IntegerType =>
+        new Converter {
+          override def set(value: Any): Unit = updater.setInt(value.asInstanceOf[IntWritable].get)
+        }
+
+      case LongType =>
+        new Converter {
+          override def set(value: Any): Unit = updater.setLong(value.asInstanceOf[LongWritable].get)
+        }
+
+      case FloatType =>
+        new Converter {
+          override def set(value: Any): Unit =
+            updater.setFloat(value.asInstanceOf[FloatWritable].get)
+        }
+
+      case DoubleType =>
+        new Converter {
+          override def set(value: Any): Unit =
+            updater.setDouble(value.asInstanceOf[DoubleWritable].get)
+        }
+
+      case StringType =>
+        new Converter {
+          override def set(value: Any): Unit =
+            updater.set(UTF8String.fromBytes(value.asInstanceOf[Text].copyBytes))
+        }
+
+      case BinaryType =>
+        new Converter {
+          override def set(value: Any): Unit = {
+            val binary = value.asInstanceOf[BytesWritable]
+            val bytes = new Array[Byte](binary.getLength)
+            System.arraycopy(binary.getBytes, 0, bytes, 0, binary.getLength)
+            updater.set(bytes)
+          }
+        }
+
+      case DateType =>
+        new Converter {
+          override def set(value: Any): Unit =
+            updater.set(DateTimeUtils.fromJavaDate(value.asInstanceOf[DateWritable].get))
+        }
+
+      case TimestampType =>
+        new Converter {
+          override def set(value: Any): Unit =
+            updater.set(DateTimeUtils.fromJavaTimestamp(value.asInstanceOf[OrcTimestamp]))
+        }
+
+      case DecimalType.Fixed(precision, scale) =>
+        new Converter {
+          override def set(value: Any): Unit = {
+            val decimal = value.asInstanceOf[HiveDecimalWritable].getHiveDecimal()
+            val v = Decimal(decimal.bigDecimalValue, decimal.precision(), decimal.scale())
+            v.changePrecision(precision, scale)
+            updater.set(v)
+          }
+        }
+
+      case st: StructType =>
+        new Converter {
+          override def set(value: Any): Unit = {
+            val orcStruct = value.asInstanceOf[OrcStruct]
+            val mutableRow = new SpecificInternalRow(st)
+            val fieldConverters: Array[Converter] = st.zipWithIndex.map { case (f, ordinal) =>
+              if (missingColumnNames.contains(f.name)) {
```

Comment:
again, why do we need to handle missing columns for nested fields?

Reply:
Okay. I'll remove this.